sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -85,10 +88,20 @@ case class Generate(
 
     if (join) child.output ++ qualified else qualified
   }
+
+  override protected def otherJsonValues: JValue = {
```

Comment:
I would strongly prefer this to be done with reflection as we do in all the the other functions defined on `TreeNode`.  Then we don't have to go a implement it separately for every LogicalPlan, PhysicalPlan and Expression.